group.go

```diff
@@ -349,6 +354,39 @@ func (g *Group) GetMetadata(key string) (Datatype, uint, interface{}, error) {
 	return datatype, valueNum, value, nil
 }
 
+// GetMetadataMap returns a map[string]*GroupMetadata where key is the key of
+// each metadata added and value is an GroupMetadata struct. The map contains
+// all group metadata previously added.
+func (g *Group) GetMetadataMap() (map[string]*GroupMetadata, error) {
+	return g.GetMetadataMapWithValueLimit(nil)
+}
+
+// GetMetadataMapWithValueLimit returns a map[string]*GroupMetadata where key is the key of
+// each metadata added and value is an GroupMetadata struct. The map contains
+// all group metadata previously added.
+// The limit parameter limits the number of values returned if string or group.
+// This is helpful for pushdown of limiting metadata. If nil, value is returned
+// in full.
+func (g *Group) GetMetadataMapWithValueLimit(limit *uint) (map[string]*GroupMetadata, error) {
```

Comment:
I don't like that `limit` is an unsigned integer but I'm fine with it to maintain consistency with `Array`.